{"@class": "de.fau.cs.inf2.cthree.data.Cluster", "id": "6ee91e33e692adf5e157f5ed0f20262e", "detectedBy": ["DIFF_HIERARCHICAL", "DIFF_DBSCAN"], "members": [{"@class": "de.fau.cs.inf2.cthree.data.CodeChange", "id": "02e2cd4ab3d80855d15703c26dcf1673", "repository": "http://git.code.sf.net/p/drjava/git_repo", "fileName": "drjava/src/edu/rice/cs/drjava/model/GlobalModelIOTest.java", "commitBeforeChange": "6e00bd0fe78b22f9b84f30182d050c16081947ef", "commitAfterChange": "d6801eea8d321f3e0015cdd20a1cba75837384cb", "methodNumberBeforeChange": 31, "methodNumberAfterChange": 31, "signatureBeforeChange": "   public void testOpenMultipleFilesError()", "signatureAfterChange": "   public void testOpenMultipleFilesError()", "diff": ["-   */", "+  /** Error checking for openening multiple files checks for null and an array w/null. */", "-    catch (IOException e) {", "-      // As we hoped, the file was not found", "-    }", "-    catch (Exception e) {", "-      fail(\"Unexpectedly exception caught!\");", "-    }", "+    catch (IOException e) { /* As we hoped, the file was not found. */ }", "+    catch (Exception e) { fail(\"Unexpectedly exception caught!\"); }", "-    assertTrue(\"non-existant file\", docs == null);", "+    assertTrue(\"non-existent file\", docs == null);", "+    ", "+    _log.log(\"testOpenMultipleFilesError completed\");"]}, {"@class": "de.fau.cs.inf2.cthree.data.CodeChange", "id": "1d72ac3d1c61b9420d1c4834a21b8bad", "repository": "http://git.code.sf.net/p/drjava/git_repo", "fileName": "drjava/src/edu/rice/cs/drjava/model/GlobalModelIOTest.java", "commitBeforeChange": "4539df5303e52babbf266acd5ad130e809adc998", "commitAfterChange": "b695cac3782ec029c31eea7503f645530ef37186", "methodNumberBeforeChange": 22, "methodNumberAfterChange": 22, "signatureBeforeChange": "   public void testOpenMultipleNonexistentFiles() throws IOException", "signatureAfterChange": "   public void testOpenMultipleNonexistentFiles() throws IOException", "diff": ["+      ", "+      public void fileNotFound(File f){", "+        fileNotFoundCount++;", "+      }", "+      ", "-", "-      fail(\"IO exception was not thrown!\");", "-      // As we hoped, the file was not found", "+      fail(\"FileNotFound exception was not thrown!\");", "-    assertEquals(\"non-existant file\", doc, null);", "+    assertTrue(\"one file was opened\", doc instanceof OpenDefinitionsDocument);", "+    listener.assertFileNotFoundCount(1);"]}], "num": 12874}